vulcan/lib/server/workflows/scripts/calc_leiden.py

```diff
@@ -4,10 +4,36 @@
 scdata = sc.read(input_path('nn_anndata.h5ad'))
 use_weights = input_bool('use_weights')
 
-# umap
+# Calculate leiden clustering
 leiden_resolution = float(input_var('leiden_resolution'))
 sc.tl.leiden(scdata, resolution = leiden_resolution, key_added = "leiden", use_weights = use_weights)
 
-##### OUTPUT
-output_json(scdata.obs['leiden'].tolist(), 'leiden.json')
+### Output
 scdata.write(output_path('leiden_anndata.h5ad'))
+
+clusts = scdata.obs['leiden'].tolist()
+
+output_json(clusts, 'leiden.json')
+output_json(
+    dict(
+        [
+            str(clust),
+            str(clust)
+        ] for clust in range(max(clusts)+1)
+    )
```

Comment:
I think the tests are failing because of a missing `,` here?

Reply:
:+1: thanks!  maybe I over-talked the completeness of these bits... zero testing so far both here and the #385 scripts, so typos and incorrectly assumed python logic are erm *likely*.